File: backend/app/schemas/disease.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class DiseaseBase(BaseModel):
    name: str
    symptoms: Optional[str] = None
    severity: Optional[str] = None
    treatment: Optional[str] = None


class DiseaseCreate(DiseaseBase):
    plant_id: int


class Disease(DiseaseBase):
    id: int
    detection_date: datetime
    status: str
    plant_id: int

    class Config:
        from_attributes = True

File: frontend/src/types/disease.ts
export interface Disease {
  id: number;
  name: string;
  symptoms?: string;
  severity?: string;
  detectionDate: string;
  treatment?: string;
  status: string;
  plantId: number;
}

export interface DiseaseCreate {
  plantId: number;
  name: string;
  symptoms?: string;
  severity?: string;
  treatment?: string;
}
